### DISCUSSION TITLE
Merge with radare2

### DISCUSSION THREAD

**Discussion by throwaway1037** [rizinorg/rizin#2445] [Archive]
I apologise in advance if this may be a painful topic, but would it at all be remotely possible for rizin to merge with radare2?

The current situation is far from ideal, because both projects are very similar, yet mutually incompatible. Plus, as far as I can tell, patches are not shared between the two projects, which means they will only become more different over time and be harder to merge, if that ever became an option.

Users are now also forced to pick between the two, or run both projects in order to get the most out of the new features each tool.

May I propose either actively working with the radare2 project to solve everybody's differences and become whole again, or at least instate some ground rules to keep the option of merging in the future faesible, like not changing "r2" to "rz" in code, or introducing small or trivial incomapibilties?

The lack of inter-cooperation is also a massive duplication of effort and a waste of what could otherwise be a single, awesome project, with everybody working together again. Plus, not keeping the dialogue open reduces the chances of future cooperation.

**Comment by XVilka** [comment]
If you have some concrete patch/commit to be ported, you could suggest it here: https://github.com/rizinorg/rizin/issues/747

We went our own way for a reason, there are massive differences in how [projects are stored](https://rizin.re/posts/introducing-projects/), [commands are parsed](https://rizin.re/posts/rzshell/), signatures are handled, and now ongoing work on the next generation of the IL: https://github.com/rizinorg/rizin/blob/dev/doc/rzil.md

**Reply by throwaway1037** [reply]
That's brilliant!

My point is that everybody could enjoy these features in a single project instead of having to run two different projects in order to enjoy the unique features of each.
